File: 代码随想录训练营/day18(513).cpp
#include "iostream"
#include "vector"
#include "DataStruct.h"
using namespace std;

class Solution{
    public:
    int maxDepth = INT_MIN;
    int result = 0;
    void traversal(TreeNode* root, int depth){
        //递归结束条件当遇到子节点结束
        if (root -> left  == nullptr && root -> right == nullptr){
             if (depth > maxDepth) {
                maxDepth = depth;
                result = root->val;
            }
            return;
        }

        if(root -> left){
            depth ++;
            traversal(root -> left, depth);
            depth --;
        }
        if(root -> right){
            depth ++;
            traversal(root -> right, depth);
            depth --;
        }
        return;
    }

    int findBottomLeftValue(TreeNode* root) {
        traversal(root, 0);
        return result;
    }

};

File: 代码随想录训练营/day18(513).py
import collections
class TreeNode:
    def __int__(self, val = 0, left = None, right = None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def findBottomLeftValue(self, root):
        if root is None:
            return 0
        q = collections.deque()
        res = 0
        q.append(root)
        while q:
            size = len(q)
            for i in range(size):
                node = q.popleft()
                if i == 0:
                    result = node.val
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)

        return result
            